src/walk.rs

```diff
@@ -77,3 +81,43 @@ where
         unreachable!()
     }
 }
+
+/// Trait that provides a nth() method to any Compound with a Cardinality
+/// annotation
```

Comment:
I believe the comments are copied from the nth, this should said `a first() method`, is that correct?